src/toil/batchSystems/singleMachine.py

```diff
@@ -275,3 +319,26 @@ def acquisitionOf(self, amount):
             yield
         finally:
             self.release(amount)
+
+
+class AcquisitionTimeoutException(Exception):
```

Comment:
This should be nested in ResourcePool or renamed to ResourceAcquisitionTimeoutException. I'd prefer the former.